src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.RenameTrackingCommitter.cs

```diff
@@ -81,59 +120,20 @@ private void RenameSymbolAndApplyChanges(CancellationToken cancellationToken, ou
                 // final solution without updating the workspace, and then finally disallow
                 // cancellation and update the workspace twice.
 
-                Document document = _snapshotSpan.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
-                if (document == null)
+                // If the task was kicked off during preview and if it was cancelled out, restart it here.
+                // If it kicked off during preview and wasn't cancelled, wait and get result.
+                RenameTrackingSolutionSet renameTrackingSolutionSet;
+                if (_renameSymbolTask == null || _renameSymbolTask.IsCanceled || _renameSymbolTask.IsFaulted)
                 {
-                    Contract.Fail("Invoked rename tracking smart tag but cannot find the document for the snapshot span.");
+                    renameTrackingSolutionSet = RenameSymbolAsync(cancellationToken).WaitAndGetResult(cancellationToken);
                 }
-
-                // Get copy of solution with the original name in the place of the renamed name
-
-                var solutionWithOriginalName = CreateSolutionWithOriginalName(document, cancellationToken);
-
-                // Get the symbol for the identifier we're renaming (which has now been reverted to
-                // its original name) and invoke the rename service.
-
-                ISymbol symbol;
-                if (!TryGetSymbol(solutionWithOriginalName, document.Id, cancellationToken, out symbol))
+                else
                 {
-                    Contract.Fail("Invoked rename tracking smart tag but cannot find the symbol");
+                    renameTrackingSolutionSet = _renameSymbolTask.WaitAndGetResult(cancellationToken);
```

Comment:
if you are going to do this, you can't capture cancellation for _renameSymbolTask.